alignak/alignakobject.py

```diff
@@ -20,70 +20,161 @@
 #
 """ This module contains only a common class for all object created in Alignak: AlignakObject.
 """
-
+from __future__ import print_function
 import uuid
+import warnings
 from copy import copy
 from alignak.property import SetProp, StringProp
 
 
 class AlignakObject(object):
-    """This class provide a generic way to instantiate alignak objects.
-    Attribute are ser dynamically, whether we un-serialize them create them at run / parsing time
+    """This class provides a generic way to instantiate alignak objects.
+    Attributes are set dynamically, whether we un-serialize them create them at run / parsing time
 
     """
 
-    properties = {'uuid': StringProp(default='')}
+    name_property = 'uuid'
+    my_type = 'unset'
+
+    properties = {'uuid': StringProp()}
+    running_properties = {}
     macros = {}
 
-    def __init__(self, params=None, parsing=True):  # pylint: disable=W0613
+    def __init__(self, params=None, parsing=True, debug=False):  # pylint: disable=W0613
+        """Initialize an Alignak base object
+
+        This method will create and set an object uuid if none exists. If no uuid still exists
+        then the object is considered as a fresh new object and all the properties are initialized
+        else the properties initialization step is skipped.
+
+        Properties initialization is done by creating a new attribute for each item defined in
+        the `properties`, `running_properties` and ` macros` dictionaries. The newly created
+        attribute is valued with the default value for the corresponding entry. If no default value
+        is defined then the attribute is set as None.
+
+        If some parameters are provided, the object attributes are updated with the corresponding
+        parameters value even if those attributes do not exist previously.
 
+        If debug is set, then this method will print information about the object properties
+
+        The parsing parameter indicates that the object is created from the configuration. As such,
+        if a property exists in the parameters the corresponding Property has its
+        `configuration_set` property set to True, else it remains False. This to allow knowing if
+        a property has been valued as default or from the configuration.
+
+        :param debug: print debug information about the object properties
+        :param params: parameters used to create the object
+        :param parsing: if True, initial creation, else, object unserialization
+        """
         if params is None:
-            return
-        all_props = {}
-        all_props.update(getattr(self, "properties", {}))
-        all_props.update(getattr(self, "running_properties", {}))
+            params = {}
+
+        if debug:
+            print('AlignakObject __init__: %s, %d params' %
```

Comment:
Yes. I'd rather used a specific logger level but it may be done in a future version because it has many impacts in the classes and, more specifically, in the tests.  The solution I implemented with a debug parameter has the advantage to be easily activated for some specific targeted tests wihout having to deal with a huge bunch of logs as we have currently!

I will reference and update #442  for such a refactoring